lib/galaxy/tools/xsd/galaxy.xsd

```diff
@@ -1652,6 +1652,8 @@ module.
       </xs:element>
       <xs:element name="has_h5_keys" type="AssertHasH5Keys">
       </xs:element>
+      <xs:element name="contains_h5_keys" type="AssertContainsH5Keys">
```

Comment:
has_ and contains_ is for a user probably the same. Could we extend the current has_ functionality?

Reply:
I suppose so. I wanted to avoid needing to specify a large number of keys for complex h5 files, but given that this is only invoked for tests, I suppose test output can be controlled to keep this list reasonable.

Reply:
Also, do you mean altering the 'has_h5_keys' test? Doing so would definitely break any tests currently using it if they have groups in the h5 files.

Reply:
I was reading this as change `has_h5_keys` to not require a complete exact match. Which makes sense, "has" does not mean "has only".

Reply:
What @jxtx said is what I wanted to say :) Sorry.

Reply:
Done